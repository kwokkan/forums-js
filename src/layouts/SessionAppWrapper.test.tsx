/**
 * @jest-environment jsdom
 */
import React from "react";
import { create } from "react-test-renderer";

function TestChild() {
    return (
        <p>Test</p>
    );
}

test("Renders without session", () => {
    jest.isolateModules(() => {
        const mockSession = jest.fn((): [INextSession | null, boolean] => [null, true]);

        jest.doMock("next-auth/client", () => {
            return {
                __esModule: true,
                useSession: mockSession
            };
        });

        const { SessionAppWrapper } = require("./SessionAppWrapper");

        const tree = create(
            <SessionAppWrapper Component={TestChild} />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});

test("Renders with session", () => {
    jest.isolateModules(() => {
        const mockSession = jest.fn((): [INextSession | null, boolean] => [{
            forumsUser: {
                id: 1,
                name: "Test user",
                joinedDate: 1
            }
        }, true]);

        jest.doMock("next-auth/client", () => {
            return {
                __esModule: true,
                useSession: mockSession
            };
        });

        const { SessionAppWrapper } = require("./SessionAppWrapper");

        const tree = create(
            <SessionAppWrapper Component={TestChild} />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
