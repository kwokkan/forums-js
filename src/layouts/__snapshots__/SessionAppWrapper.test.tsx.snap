// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders with session 1`] = `
<div
  className="navbar-page"
>
  <div
    className="rs-container"
  >
    <div
      className="rs-header"
    >
      <div
        className="rs-navbar rs-navbar-default"
        role="navigation"
      >
        <div
          className="rs-navbar-header"
        >
          <a
            className="navbar-brand logo"
            href="/"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <img
              alt="Logo"
              className="mr-3"
              height="24"
              src="/logo.png"
              title="Forums JS"
              width="24"
            />
            Forums JS
          </a>
        </div>
        <div
          className="rs-navbar-body"
        >
          <div
            className="rs-nav rs-nav-default rs-navbar-nav rs-nav-horizontal"
          >
            <ul>
              <li
                className="rs-nav-item"
              >
                <a
                  className="rs-nav-item-content"
                  href="/"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  <i
                    className="rs-icon rs-icon-home"
                  />
                  Home
                  <span
                    className="rs-ripple-pond"
                  >
                    <span
                      className="rs-ripple"
                      style={Object {}}
                    />
                  </span>
                </a>
              </li>
              <li
                className="rs-nav-item"
              >
                <a
                  className="rs-nav-item-content"
                  onClick={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  Forums
                  <span
                    className="rs-ripple-pond"
                  >
                    <span
                      className="rs-ripple"
                      style={Object {}}
                    />
                  </span>
                </a>
              </li>
              <li
                className="rs-nav-item"
              >
                <a
                  className="rs-nav-item-content"
                  onClick={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  New Posts
                  <span
                    className="rs-ripple-pond"
                  >
                    <span
                      className="rs-ripple"
                      style={Object {}}
                    />
                  </span>
                </a>
              </li>
              <li
                className="rs-nav-item"
              >
                <a
                  className="rs-nav-item-content"
                  href="/about"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  About
                  <span
                    className="rs-ripple-pond"
                  >
                    <span
                      className="rs-ripple"
                      style={Object {}}
                    />
                  </span>
                </a>
              </li>
            </ul>
          </div>
          <div
            className="rs-nav rs-nav-default rs-navbar-nav rs-navbar-right rs-nav-horizontal"
          >
            <ul>
              <li
                className="rs-nav-item"
              >
                <a
                  className="rs-nav-item-content"
                  href="https://github.com/kwokkan/forums-js"
                  onClick={[Function]}
                  rel="noreferrer noopener"
                  role="button"
                  tabIndex={0}
                  target="_blank"
                >
                  <i
                    className="rs-icon rs-icon-github"
                  />
                  Fork me
                  <span
                    className="rs-ripple-pond"
                  >
                    <span
                      className="rs-ripple"
                      style={Object {}}
                    />
                  </span>
                </a>
              </li>
              <li
                className="rs-nav-item"
              >
                <a
                  className="rs-nav-item-content"
                  href="/settings"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  <i
                    className="rs-icon rs-icon-cog"
                  />
                  Settings
                  <span
                    className="rs-ripple-pond"
                  >
                    <span
                      className="rs-ripple"
                      style={Object {}}
                    />
                  </span>
                </a>
              </li>
              <li
                className="rs-nav-item"
              >
                <a
                  className="rs-nav-item-content"
                  onClick={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  <i
                    className="rs-icon rs-icon-cog"
                  />
                  Logout
                  <span
                    className="rs-ripple-pond"
                  >
                    <span
                      className="rs-ripple"
                      style={Object {}}
                    />
                  </span>
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
    <div
      className="rs-content"
    >
      <div
        className="rs-message rs-message-info rs-message-show rs-message-has-title rs-message-has-icon"
      >
        <div
          className="rs-message-container"
        >
          <div
            className="rs-message-icon-wrapper"
          >
            <i
              className="rs-icon rs-icon-info"
            />
          </div>
          <div
            className="rs-message-content"
          >
            <h5
              className="rs-message-header"
            >
              Coming soon
            </h5>
            <div
              className="rs-message-body"
            >
              <p>
                Coming soon.
              </p>
            </div>
          </div>
        </div>
      </div>
      <p>
        Test
      </p>
    </div>
  </div>
</div>
`;

exports[`Renders without session 1`] = `
<div
  className="navbar-page"
>
  <div
    className="rs-container"
  >
    <div
      className="rs-header"
    >
      <div
        className="rs-navbar rs-navbar-default"
        role="navigation"
      >
        <div
          className="rs-navbar-header"
        >
          <a
            className="navbar-brand logo"
            href="/"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <img
              alt="Logo"
              className="mr-3"
              height="24"
              src="/logo.png"
              title="Forums JS"
              width="24"
            />
            Forums JS
          </a>
        </div>
        <div
          className="rs-navbar-body"
        >
          <div
            className="rs-nav rs-nav-default rs-navbar-nav rs-nav-horizontal"
          >
            <ul>
              <li
                className="rs-nav-item"
              >
                <a
                  className="rs-nav-item-content"
                  href="/"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  <i
                    className="rs-icon rs-icon-home"
                  />
                  Home
                  <span
                    className="rs-ripple-pond"
                  >
                    <span
                      className="rs-ripple"
                      style={Object {}}
                    />
                  </span>
                </a>
              </li>
              <li
                className="rs-nav-item"
              >
                <a
                  className="rs-nav-item-content"
                  onClick={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  Forums
                  <span
                    className="rs-ripple-pond"
                  >
                    <span
                      className="rs-ripple"
                      style={Object {}}
                    />
                  </span>
                </a>
              </li>
              <li
                className="rs-nav-item"
              >
                <a
                  className="rs-nav-item-content"
                  onClick={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  New Posts
                  <span
                    className="rs-ripple-pond"
                  >
                    <span
                      className="rs-ripple"
                      style={Object {}}
                    />
                  </span>
                </a>
              </li>
              <li
                className="rs-nav-item"
              >
                <a
                  className="rs-nav-item-content"
                  href="/about"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  About
                  <span
                    className="rs-ripple-pond"
                  >
                    <span
                      className="rs-ripple"
                      style={Object {}}
                    />
                  </span>
                </a>
              </li>
            </ul>
          </div>
          <div
            className="rs-nav rs-nav-default rs-navbar-nav rs-navbar-right rs-nav-horizontal"
          >
            <ul>
              <li
                className="rs-nav-item"
              >
                <a
                  className="rs-nav-item-content"
                  href="https://github.com/kwokkan/forums-js"
                  onClick={[Function]}
                  rel="noreferrer noopener"
                  role="button"
                  tabIndex={0}
                  target="_blank"
                >
                  <i
                    className="rs-icon rs-icon-github"
                  />
                  Fork me
                  <span
                    className="rs-ripple-pond"
                  >
                    <span
                      className="rs-ripple"
                      style={Object {}}
                    />
                  </span>
                </a>
              </li>
              <li
                className="rs-nav-item"
              >
                <a
                  className="rs-nav-item-content"
                  href="/api/auth/signin"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  <i
                    className="rs-icon rs-icon-cog"
                  />
                  Login
                  <span
                    className="rs-ripple-pond"
                  >
                    <span
                      className="rs-ripple"
                      style={Object {}}
                    />
                  </span>
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
    <div
      className="rs-content"
    >
      <div
        className="rs-message rs-message-info rs-message-show rs-message-has-title rs-message-has-icon"
      >
        <div
          className="rs-message-container"
        >
          <div
            className="rs-message-icon-wrapper"
          >
            <i
              className="rs-icon rs-icon-info"
            />
          </div>
          <div
            className="rs-message-content"
          >
            <h5
              className="rs-message-header"
            >
              Coming soon
            </h5>
            <div
              className="rs-message-body"
            >
              <p>
                Coming soon.
              </p>
            </div>
          </div>
        </div>
      </div>
      <p>
        Test
      </p>
    </div>
  </div>
</div>
`;
